src/avalanchemq/http/handler/basic_auth.cr

```diff
@@ -58,6 +58,11 @@ module AvalancheMQ
         return false if user.password.to_s.empty?
         user.password.not_nil!.verify(password)
       end
+
+      private def guest_localhost?(context, user)
+        return true unless user.name == "guest"
```

Comment:
Should we make it configurable? or is it good enough to hardcode at this point?

Reply:
Maybe it could be nice to override, just realised I actually did use `guest/guest` over network, but a private network. (used my Tailscale cert to test HTTPS/AMQPS on my machine https://tailscale.com/blog/tls-certs/).

... or make it easier to get a non-guest default user (create random username/password, print URL to stdout for quick usage).

Will we block easy Docker test usage? (not localhost).

</rambling>

Reply:
Yes, been thinking the same, witht the random password for default user. And yes, the current hard block first for a non local connections is a blocker for docker adaption. Rabbitmq disables the protection in their image, and allow guest

Reply:
So at least make the check enabled/disabled in the configuration? Looking in to the user store, as for now there is no way to detect a user as default what I can see. Could another tag or field be used for this?

Reply:
> So at least make the check enabled/disabled in the configuration? 

Yes, and default disabled. Guess a config file and CLI option would be good (CLI option could be used for container).

> Looking in to the user store, as for now there is no way to detect a user as default what I can see. Could another tag or field be used for this?

Correct, It's "dynamic", first user with tag Admin IIRC. There was a field in `VHost` that I removed recently since it wasn't updated correctly see https://github.com/cloudamqp/avalanchemq/pull/290 and https://github.com/cloudamqp/avalanchemq/issues/276 

Yes maybe we need to rethink that, not sure what's the best way forward.

Feel free to split up into two PRs, i.e. block guest/guest (with option to disable) here, then another PR that will create random the password (or username + password).